lib/mayaUsd/ufe/ProxyShapeSceneSegmentHandler.cpp

```diff
@@ -45,6 +45,41 @@ ProxyShapeSceneSegmentHandler::create(const Ufe::SceneSegmentHandler::Ptr& mayaS
 // Ufe::SceneSegmentHandler overrides
 //------------------------------------------------------------------------------
 
+#if (UFE_PREVIEW_VERSION_NUM >= 4033)
+Ufe::Selection
+ProxyShapeSceneSegmentHandler::findGatewayItems_(const Ufe::Path& path, Ufe::Rtid nestedRtid) const
+{
+    Ufe::Selection result = Ufe::Selection();
+
+    // Handle other gateway node types that MayaUSD is not aware of.
+    // `nestedRtid` is used as a filter. If it matches the MayaUSD runtime ID, MayaUSD is aware of
+    // the requested gateway items so this step can be skipped.
+    if (nestedRtid != g_USDRtid && fMayaSceneSegmentHandler) {
+        result = fMayaSceneSegmentHandler->findGatewayItems_(path, nestedRtid);
```

Comment:
Based on the two conditions used in this method's implementation, it seems that the code can directly return.

Reply:
No, I don't think so. If `nestedRtid` is `Ufe::kAllRtid`, the method has to do both: Handle gateways that MayaUSD is not aware of (this if block) **and** find the MayaUSD proxyShapes (next if block).